### ISSUE TITLE
[ENHANCEMENT] Make "Absorb Heat" smarter (faster/shorter)

### ISSUE THREAD

**Issue by MR2C280** [prusa3d/Prusa-Firmware-Buddy#3534]
The "Absorb Heat" phase of printing adds several minutes to the print time and is a bit of a 'blunt instrument'.  I think it can be smarter.

I assume the absorb heat phase to be sure the print bed is fully up to temp and stable before printing begins.  But it does this on every print without taking the conditions into consideration.  For example: Is the printer already warm?  Has the user preheated the bed?  Is the object being printed prone to bed adhesion issues (a big flat square surface vs. many small islands for organic supports)?  How long will it take to get all the print heads heated & primed, nozzles cleaned and mesh bed levelling probed? (the bed temp will be _absorbing_ during that time)

Possible solutions:
1) Add an Absorb Heat time setting in the Filament tab in PrusaSlicer (perhaps with an override on the Print tab).
2) Consider the conditions mentioned above to adjust the time accordingly.
3) Monitor the adjacent unheated bed segments for either a specific temperature or for when their temps stops rising.

I know this applies to the XL bit may also apply to MINI and MK4 too.

Personally, I have commented out the **G29 G ; absorb heat** in the Printer Start G-code as I find it is (usually) unnecessary.  But, as I said, that's a bit of a blunt instrument and sometimes come to regret not being a little more patient.

**Comment by brettvitaz** [comment]
> Allow the user to skip or shorten the Absorb Heat phase via the LCD

This already exists. You just select "skip" from the printer screen when it is in the absorb heat phase.

**Comment by DoubleStrike** [comment]
> > Allow the user to skip or shorten the Absorb Heat phase via the LCD
> 
> This already exists. You can select "skip" from the printer screen when it is in the absorb heat phase.

No, this doesn't exist.  What OP is asking for is more intelligence in when heat absorbing is applied, and to how much time it calculates.  Having to manually skip it is not optimal, as there is a reason for it to exist, but it just needs to be smarter.

**Comment by DanielWeigl** [comment]
> This already exists. You can select "skip"

Not while printing via Octoprint 

Maybe an easier idea would be to take the medium temperature before the "Absorb Heat" step starts and use the difference to the target temperature as basis for the time it should wait - so, if its already preheated, it shouldnt take that long

**Comment by MR2C280** [comment]
This may not be a high priority but please don't close the issue.

**Comment by Racemade** [comment]
+1

**Comment by AccaliaDeElementia** [comment]
Agreed on this issue. it's weird that there'sa  difference in the behavior of the absorb heat gcode when printing from SD/USB drive verses when printing from Serial.

It would be nice if the behavior was consistent between the two and didnt' generate a 10 minute hang in printing when using octoprint. Yes the prints are almost always successful omitting the absorb heat code, but having access to the gcode would make for a better experience.

**Comment by nkiest** [comment]
Also, if triggering prints remotely, you can't make it skip. I don't really know why Prusa Connect has a preheat function if it doesn't reduce absorb time.

**Comment by foorschtbar** [comment]
bump up. Still no option to skip this step with OctoPrint - and yes Prusa, people use OctoPrint and want this also in the future.

**Comment by hagre** [comment]
One "easy" to implement solution could be to modyfy the existing time-calculation in the Prusa-Firmware-Buddy/lib/Marlin/Marlin/src/feature/bed_preheat.cpp

uint32_t BedPreheat::required_preheat_time() {
    if (thermalManager.degTargetBed() < minimal_preheat_temp) {
        return 0;
    }
    return std::max((180 + (thermalManager.degTargetBed() - 60) * (12 * 60 / 50)) * 1000, 0);
}

and use the actual bed temperature (std::abs(thermalManager.degBed()) to reduce the calculated time if already preheated. Caution: On XL modular bed, this could maybe use average temperature of a different segment.

**Comment by DmetlVtoore** [comment]
I had issues with this too.  I frequently do small prints of 15 to 30 minutes in PETG and the additional 7 minutes is overkill for these prints.

A simple solution for me would be to do a linear interpolation based on a fixed print time.  For example, with a full absorption time of 7 minutes (for PETG) any print 2 hours or longer would do the full absorption time.  A print of 1 hour would do half absorption time and a 20 minute print would add only 70s [7m * 20m / 120m].

**Comment by cerionig** [comment]
+1 However, in my case I would like to extend the preheating, but only for materials like ASA that benefit from a warmer chamber, especially on cold days / first print of the day.

**Comment by SetupCoding** [comment]
+1 Enabling Skip in either OctoPrint or in the PrusaSlicer is a must-have.

**Comment by EvilVir** [comment]
I would like to have ability to disable this annoying functionality.

**Comment by MR2C280** [comment]
> I would like to have ability to disable this annoying functionality.

You can disable the absorb heat function by removing it from the Start G-code

<img width="1177" height="541" alt="Image" src="https://github.com/user-attachments/assets/923f6f42-ed67-46d9-a8f8-c67a4d49896b" />

**Comment by Crocmagnon** [comment]
I believe « absorbing heat » is for enclosed printers printing PLA/PETG which prints best at room temperature so it turns on the enclosure fans to (try to) reduce the temperature in the enclosure. It’s not about bed or nozzle temp as far as I understand the feature.
If you turn on the enclosure temp in the footer of the LCD screen, you can see during this phase that it blinks blue, indicating a cooldown.

**Comment by morphias2004** [comment]
@Crocmagnon you are factually incorrect in almost the entirety of your response. Reading the manual for the XL will confirm this.

Opening a statement like that with "I believe" is poor form. Please don't add uneducated/uninformed information. It adds nothing but confusion and conjecture.

**Comment by Crocmagnon** [comment]
@morphias2004 please point to official documentation. The [XL handbook](https://www.prusa3d.com/downloads/manual/prusa3d_manual_xl_105_en.pdf) has no reference for "absorbing heat", same for the [CORE One handbook](https://www.prusa3d.com/downloads/manual/prusa3d_manual_coreone_101_en.pdf).
There are [no relevant search result](https://help.prusa3d.com/search/absorbing%20heat) for this term on the support site either.
There are references to "active temperature control" in the [marketing](https://blog.prusa3d.com/introducing-prusa-core-one-fully-enclosed-corexy-3d-printer-with-active-temperature-control_105477/) [material](https://www.prusa3d.com/en/product/prusa-core-one/) though.

**Comment by morphias2004** [comment]
@Crocmagnon OK, so stupidly not in the manual. Do better @Prusa-Support 

It is covered here:

https://blog.prusa3d.com/original-prusa-xl-now-shipping_75721/

<img width="990" height="1324" alt="Image" src="https://github.com/user-attachments/assets/b197727a-621c-4d00-9388-d7dd025847ad" />

And is also covered in multiple Prusa Forum discussions as well as plenty of other places.

https://forum.prusa3d.com/forum/original-prusa-xl-tool-changer-hardware-firmware-and-software-help/insight-on-how-heat-absorption-is-calculated/

https://forum.prusa3d.com/forum/original-prusa-xl-tool-changer-assembly-and-first-prints-troubleshooting/why-the-unnecessary-waits-before-printing/

https://forum.prusa3d.com/forum/original-prusa-xl-tool-changer-assembly-and-first-prints-troubleshooting/absorbing-heat-before-printing/

**Comment by Crocmagnon** [comment]
Interesting thanks, I wonder then why the core one has this step too. Maybe they used the same wording for two different features in two different machines?
Because when "absorbing heat" is displayed on the core one, the chamber fans (or the HEPA filter fans) spin up and the chamber temp blinks blue.

**Comment by garrickp** [comment]
+1

The reason it exists makes sense, but the way it works doesn't jive with that reason. And agressively incorrect comments don't help.

**Comment by hyiger** [comment]
I think it would be clearer to display "Heat Bed Temperature Stabilization" or something like that. Also, provide a way to turn it off in the firmware instead of having to comment out the g-code sent to the printer.

**Comment by foorschtbar** [comment]
> Thank you for your contribution to our project. This issue has not received any updates for 60 days and may be considered "stale." If this issue is still important to you, please add an update within the next 7 days to keep it open. Administrators can manually reopen the issue if necessary.

no stale, still wait for an option to control the Absorb heat from the Slicer and a way to skip it when using OctoPrint.

**Comment by mhouse1** [comment]
+1 would be nice to be able to turn absorb heat on or off via touch UI, or configure it for specific filaments.

**Comment by 1c3Panda** [comment]
+1 to keep this alive - would love to have an option in the slicer to allow control when remote/serial printing

**Comment by artemonic** [comment]
I think this one needs a better solution or explanation.
I mostly print PETG or PLA, occasionally ASA. Built my Core One a month ago, so pretty new to it.
I noticed the filaments profiles have the Chamber temperature added with 'nominal' and 'minimal' parameters. The 'nominal' numbers is what I see active on the screen during the 'Absorbing Heat' phase. For the PETG it's 35°C for example, which doesn't make much sense, but please correct ne if I'm wrong.
I also see some smarter solutions around the community, like [this comment on Prusa's forum](https://forum.prusa3d.com/forum/postid/763311/)
```
G28 ; home all without mesh bed level

;
; Added - No chamber heater for PLA/PETG
;
{if chamber_minimal_temperature[initial_tool] > 36}
; CHAMBER HEATING AND WAIT FOR MATERIALS THAT NEED >36 °C CHAMBER
M104 S{idle_temperature[initial_tool]} ; set idle temp
G1 Z10 F720 ; set bed position
G1 X242 Y-9 F4800 ; set print head position
M191 S{chamber_minimal_temperature[initial_tool]} ; wait for minimal chamber temp
M141 S{chamber_temperature[initial_tool]} ; set nominal chamber temp
M107
M140 S[first_layer_bed_temperature] ; set bed temp
{else}
; SKIP CHAMBER HEATING FOR MATERIALS WITH MINIMAL CHAMBER TEMP ≤ 36 °C (E.G. PLA, PETG)
M141 S0 ; disable chamber heater
{endif}

{if first_layer_bed_temperature[initial_tool]<=60}M106 S70{endif}
G0 Z40 F10000
M104 T{initial_tool} S{if is_nil(idle_temperature[initial_tool])}100{else}{idle_temperature[initial_tool]}{endif}
M190 R[first_layer_bed_temperature] ; wait for bed temp
M107

;
; Added - No soak for PLA/PETG
;
{if chamber_minimal_temperature[initial_tool] > 36}
G29 G ; ABSORB HEAT FOR HIGH-TEMP MATERIALS
{else}
G4 S5 ; SHORT PAUSE INSTEAD OF SOAK FOR LOW-TEMP MATERIALS
 #{endif}
```

or [this one](https://www.reddit.com/r/prusa3d/comments/1ilu19b/comment/nl50sk5/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button)

```
; === OVERRIDE PRUSA SETTINGS - nominal chamber temp ===
; Nominal or "Max Chamber Temperature" from Tuning menu to 55c if filament
; type Starts With abs/asa/pc/pa, or Ends With cf/gf
{if (filament_type[initial_tool] =~ /^(ABS|ASA|PC|PA)/) or (filament_type[initial_tool] =~ /(CF|GF)$/)}
    M141 S55 ; set nominal chamber temp
{endif}
```


But since I don't understand it fully, it would be nice to have some more confirmation on this solution

**Comment by mhouse1** [comment]
```
;G29 G ; absorb heat
```

Our current approach comments out G29 G entirely in all CORE One profiles. The farm runs predominantly PETG, where the thermal expansion delta is small enough that the absorb heat dwell adds wait time with minimal quality benefit — and at farm scale that cost compounds across every print start.

The limitation is that this decision is made at slice time and locked into the binary file. A material-conditional workaround (gating on chamber_minimal_temperature) improves on this by enabling absorb heat automatically for ABS/ASA, but it still can't account for factors only visible at print time — a large flat PETG part with high bed contact area, a cold-start after overnight idle, or operator judgement on a borderline job. Both approaches are ultimately workarounds in start G-code.

The right solution is a runtime toggle in the printer UI. That would let an operator make the call per print based on geometry, thermal state, and material — without re-slicing. Until Prusa ships that, commenting out G29 G is the pragmatic choice for a PETG-heavy farm, with the understanding that ABS jobs and large flat parts are leaving some first-layer consistency on the table.

**Comment by mbj** [comment]
The firmware could maintain a "thermal history" and dynamically adjust the absorbtion timer. Lets say you come out of a 2h print, and start the next print within 5min, the printer could adjust the absorbtion timeline proportional to that history as it knows that the entire frame is properly pre-heated?

**Comment by obrienn** [comment]
On 6.5.3+12780 commenting out G29 G no longer works by itself for. This holds true when setting the filament minimum chamber temp to 0. It only seems to respect not waiting for chamber temps if the nominal and minimum chamber temps are 0, and the G29 G line is commented out.

**Comment by zzqzzqzzq** [comment]
Pretty sure its not stale, unless Prusa is considering their GEN2 upgrade its fix....

**Comment by mhouse1** [comment]
Anyone try the Gen2 upgrade, does it resolve it? I'd rather not have to add custom GCode to every object, it just seems to make sense for the printer to auto handle this.
